BlueprintUI/Sources/Layout/Stack.swift

```diff
@@ -70,14 +70,40 @@ extension StackElement {
     ///                     }
     ///                     ```
     ///
+    ///   - alignmentGuide: A closure that can be used to provide a custom alignment guide for this
+    ///                     child.
+    ///
+    ///                     This closure will be called with an `ElementDimensions` containing the
+    ///                     dimensions of this child, and should return a value in the child's
+    ///                     own coordinate space. This value represents the position along the
+    ///                     child's cross axis that should be aligned relative to the values of
+    ///                     its siblings.
+    ///
+    ///                     If not specified, the default value is provided by the stack's
+    ///                     alignment.
+    ///
+    ///                     The alignment guide is ignored by the `fill` alignment type.
+    ///
     ///   - key: A key used to disambiguate children between subsequent updates of the view hierarchy
     ///
     ///   - child: The child element to add to this stack
     ///
-    public mutating func add(growPriority: CGFloat = 1.0, shrinkPriority: CGFloat = 1.0, key: AnyHashable? = nil, child: Element) {
+    /// - Tag: StackElement.add
+    ///
+    mutating public func add(
+        growPriority: CGFloat = 1.0,
+        shrinkPriority: CGFloat = 1.0,
+        alignmentGuide: ((ElementDimensions) -> CGFloat)? = nil,
```

Comment:
Because you usually build the element inline, so it looked weird having little "parameter turds" hanging off the end like this:

```
stack.add(
child: MyElement(
   thatWrap: AnotherThing(
     with: more,
     params: likeThis
   ) { 
     $0.configuring...
  }
  growPriority: 1,
  shrinkPriority: 1,
)
```

Reply:
https://github.com/square/Blueprint/issues/17